include/ares.h

```diff
@@ -345,10 +346,16 @@ struct apattern;
  * to.
  * The retry delay is the minimum time in milliseconds to wait between doing
  * such retries (applied per-server).
+ * The min_consec_successes is the minimum number of consecutive successful
+ * queries before a server is considered "up" again.
+ * The max_consec_failures is the maximum number of consecutive failed
+ * queries before a server is considered "down".
  */
 struct ares_server_failover_options {
   unsigned short retry_chance;
   size_t         retry_delay;
+  size_t         min_consec_successes;
+  size_t         max_consec_failures;
 };
```

Comment:
this is sort of an odd approach to add a new ARES_OPT but then modify an existing option.  I think I'd rather just a setter/getter approach rather than using ares_init_options() for this.  Infact, I hope to remove ares_init_options() one day.